hot-reload-agent/src/main/kotlin/org/jetbrains/compose/reload/agent/reload.kt

```diff
@@ -109,10 +113,33 @@ internal fun reload(
         ClassDefinition(originalClass, transformed)
     }
 
+    val changedResources = pendingChanges.keys.filter { !it.isClass() && it.isFile }
+
+    if (changedResources.isNotEmpty()) {
+        cleanResourceCache()
+    }
+
     instrumentation.redefineClasses(*definitions.toTypedArray())
-    return redefineRuntimeInfo().get().mapLeft { redefinition ->
+    return redefineRuntimeInfo(changedResources).get().mapLeft { redefinition ->
         val reload = Reload(reloadRequestId, definitions, redefinition)
         reinitializeStaticsIfNecessary(reload)
         reload
     }
 }
+
+private fun cleanResourceCache(classId: ClassId, cleanMethod: MethodId, resourceCacheType: String) {
+    val loader = findClassLoader(classId).get()
+    if (loader == null) {
+        logger.info("$resourceCacheType resource cache cleaning skipped: '$classId' is not loaded yet.")
+        return
+    }
+    loader.loadClass(classId.toFqn())
+        .getDeclaredMethod(cleanMethod.methodName)
+        .invoke(null)
+    logger.info("$resourceCacheType resource cache cleared")
```

Comment:
I would propose to wrap this in. `Try {}`:

A failure in resource invalidation shall be reported aggressively, but not break the reload. 
E.g. when compose resources breaks this integration in later changes, then we should not fail the reload, but report that the integration to compose resources is not working.

Reply:
It is already wrapped in `Try {}` at the usage.